src/main/java/emissary/core/BaseDataObject.java

```diff
@@ -462,6 +462,16 @@ public void setData(@Nullable final byte[] newData, final int offset, final int
         safeUsageChecker.resetCacheThenRecordSnapshot(theData);
     }
 
+    /**
+     * Checks if the data is defined with a non-zero length.
+     * 
+     * @return if data is undefined or zero length.
+     */
+    @Override
+    public boolean isDataEmpty() throws IOException {
+        return getDataState() == DataState.NO_DATA || getChannelSize() == 0;
+    }
```

Comment:
Could we invert this logic and change its name to `hasContent`?  That name seems more intuitive to me, and it keeps the internals internal. (`getChannelSize` already includes the `getDataState()` check, so its inclusion here would be redundant).
```suggestion
    public boolean hasContent() throws IOException {
        return getChannelSize() > 0;
    }
```